### PR TITLE
🍔 Added the Jenkins Custom Build stuff 🍔

### PR DESCRIPTION
**Pull request by springdo**
🍔 Custom build jenkins stuff, secret for pulling against a protected end point. Stuff you would have found in what we were doing before, but now with 100% more helm

https://github.com/redhat-cop/openshift-templates/tree/master/jenkins-s2i-build

### CODE REVIEW

File: jenkins/templates/external-jenkins-imagestream.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{{- if .Values.imagestream.external }}
+---
+kind: ImageStream
+apiVersion: image.openshift.io/v1
+metadata:
+  name: {{ .Values.imagestream.name }}
```
[the remark this answers is not in the document]
Comment (ckavili) [reply jenkins/templates/external-jenkins-imagestream.yaml]: I guess the problem was to convert int value to string for builder_imagetag so I added quote for it. It should be fine now.
```
name: {{ .Values.imagestream.builder_imagetag | quote }}
```

File: jenkins/values.yaml [hunk 2]
```diff
@@ -63,3 +69,23 @@ services:
...
+    builder_imagetag: latest
+  builder_repo: openshift
+  builder_image: jenkins
+  builder_imagetag: '2'
+
+buildconfig:
+  name: *name
+  build_secret: '🔥shhhhhhh-this-is-my-super-duper-secret123-shhhhhhh🔥'
+  source_secret: 'git-auth'
+  pull_secret: 'biscuits-123'
```
[the remark this answers is not in the document]
Comment (ckavili) [reply jenkins/values.yaml]: You're right, updated as blank 😅

File: jenkins/values.yaml [hunk 3]
```diff
@@ -3,51 +3,57 @@ appName: &name jenkins
...
+  password: thisisdefinitelymypassword
+
 deployment:
   openshiftauth: true
   imagestream:
-    name: jenkins
+    name: *name
     tag: latest
-    namespace: openshift
+    namespace: labs-ci-cd
```
[the remark this answers is not in the document]
Comment (ckavili) [reply jenkins/values.yaml]: Also Argo knows where to deploy..soo removed it 👍🏻

File: jenkins/templates/buildconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      uri: {{ .Values.buildconfig.source_repo_url | quote }}
+      ref: {{ .Values.buildconfig.source_repo_ref | default "master" | quote }}
+    contextDir: {{ .Values.buildconfig.source_repo_context | quote }}
+  strategy:
+    type: Source
+    sourceStrategy:
+      from:
+        kind: ImageStreamTag
+        name: "{{ .Values.imagestream.name }}:{{ .Values.imagestream.builder_imagetag }}"
+        namespace: {{ .Values.imagestream.builder_repo }}
```
[the remark this answers is not in the document]
Comment (ckavili) [reply jenkins/templates/buildconfig.yaml]: I didn't know it automatically adds current namespace value here. Removed it also 🌿

File: jenkins/templates/external-jenkins-imagestream.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ckavili) [reply jenkins/templates/external-jenkins-imagestream.yaml]: > 🎠🎠 @springdo, @ckavili - when deploying to ocp4.3 i also had to do following 🎠🎠
> 
> ClusterRole edit needs to be bound locally to jenkins_edit SA .. there is no local edit role in namespace
> 

You're right, it should be ClusterRole. Updated it. Honestly I don't know much about other repos but let me check with @springdo :) And thanks for the comments 😇

**Comment by eformat** [comment]
LGTM

**Pull request merged**